packages/test/test-end-to-end-tests/src/test/incrementalSummaryContext.spec.ts

```diff
@@ -0,0 +1,617 @@
+		throw new Error("Method not implemented.");
+	}
+}
+
+/**
+ * Validates that incremental summaries can be created at the sub DDS level
+ */
+describeNoCompat("Incremental summaries can be generated for DDSes", (getTestObjectProvider) => {
+	let provider: ITestObjectProvider;
+	const dataObjectFactory = new TestFluidObjectFactory([
+		[
+			TestIncrementalSummaryTreeDDS.getFactory().type,
+			TestIncrementalSummaryTreeDDS.getFactory(),
+		],
+		[
+			TestIncrementalSummaryBlobDDS.getFactory().type,
+			TestIncrementalSummaryBlobDDS.getFactory(),
+		],
+	]);
+	const runtimeOptions: IContainerRuntimeOptions = {
+		summaryOptions: { summaryConfigOverrides: { state: "disabled" } },
+	};
+	const innerRequestHandler = async (request: IRequest, runtime: IContainerRuntimeBase) =>
+		runtime.IFluidHandleContext.resolveHandle(request);
+	const runtimeFactory = new ContainerRuntimeFactoryWithDefaultDataStore(
+		dataObjectFactory,
+		[[dataObjectFactory.type, Promise.resolve(dataObjectFactory)]],
+		undefined,
+		[innerRequestHandler],
+		runtimeOptions,
+	);
+
+	const createContainer = async (): Promise<IContainer> => {
+		return provider.createContainer(runtimeFactory);
+	};
+
+	async function loadContainer(summaryVersion: string) {
+		return provider.loadContainer(runtimeFactory, undefined, {
+			[LoaderHeader.version]: summaryVersion,
+		});
+	}
+
+	async function createSummarizer(container: IContainer, summaryVersion?: string) {
+		const createSummarizerResult = await createSummarizerFromFactory(
+			provider,
+			container,
+			dataObjectFactory,
+			summaryVersion,
+			getContainerRuntimeApi(pkgVersion, pkgVersion)
+				.ContainerRuntimeFactoryWithDefaultDataStore,
+		);
+		return createSummarizerResult.summarizer;
+	}
+
+	beforeEach(async () => {
+		provider = getTestObjectProvider({ syncSummarizer: true });
+	});
+
+	it("can create summary handles for blobs in DDSes that do not change", async () => {
+		const container = await createContainer();
+		const datastore = await requestFluidObject<ITestFluidObject>(container, "default");
+		const dds = await datastore.getSharedObject<TestIncrementalSummaryBlobDDS>(
+			TestIncrementalSummaryBlobDDS.getFactory().type,
+		);
+		// Each op leads to the creation of a new blob in the summary.
+		// Older blobs never are modified in this case for simplicity sake.
+		dds.createBlobOp("test data 1");
+		dds.createBlobOp("test data 2");
+		dds.createBlobOp("test data 3");
+
+		const summarizer = await createSummarizer(container);
+		await provider.ensureSynchronized();
+		await summarizeNow(summarizer);
+
+		// This op goes into a different blob. The previous unchanged 3 should be summarized as summary handles.
+		dds.createBlobOp("test data 4");
+
+		await provider.ensureSynchronized();
+		const { summaryTree } = await summarizeNow(summarizer);
+
+		// Verify the summary tree is generated as we expected it to be.
+		assert(summaryTree.tree[".channels"].type === SummaryType.Tree, "expecting a tree!");
```

Comment:
Add messages to all the asserts so that if the test fails in CI, its possible to tell which assert failed.